src/m365/viva/commands/engage/engage-community-add.ts

```diff
@@ -0,0 +1,220 @@
+import { Logger } from '../../../../cli/Logger.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { validation } from '../../../../utils/validation.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  displayName: string;
+  description: string;
+  privacy: string;
+  adminEntraIds?: string;
+  adminEntraUserNames?: string;
+}
+
+class VivaEngageCommunityAddCommand extends GraphCommand {
+  private readonly privacyOptions = ['public', 'private'];
+
+  public get name(): string {
+    return commands.ENGAGE_COMMUNITY_ADD;
+  }
+
+  public get description(): string {
+    return 'Creates a new community in Viva Engage';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        displayName: args.options.displayName !== 'undefined',
+        description: typeof args.options.description !== 'undefined',
+        privacy: typeof args.options.privacy !== 'undefined',
+        adminEntraIds: typeof args.options.adminEntraIds !== 'undefined',
+        adminEntraUserNames: typeof args.options.adminEntraUserNames !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      { option: '--displayName <displayName>' },
+      { option: '--description <description>' },
+      {
+        option: '--privacy <privacy>',
+        autocomplete: this.privacyOptions
+      },
+      { option: '--adminEntraIds [adminEntraIds]' },
+      { option: '--adminEntraUserNames [adminEntraUserNames]' },
+      { option: '--wait' }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.displayName && args.options.displayName.length > 255) {
+          return `The maximum amount of characters for 'displayName' is 255.`;
+        }
+
+        if (args.options.description && args.options.description.length > 1024) {
+          return `The maximum amount of characters for 'description' is 1024.`;
+        }
+
+        if (args.options.privacy && this.privacyOptions.indexOf(args.options.privacy) === -1) {
+          return `'${args.options.privacy}' is not a valid value for privacy. Allowed values are: ${this.privacyOptions.join(', ')}.`;
+        }
+
+        if (args.options.adminEntraIds && args.options.adminEntraUserNames) {
+          return `You can only specify either 'adminEntraIds' or 'adminEntraUserNames'`;
+        }
+
+        if (args.options.adminEntraIds) {
+          const adminEntraIds = args.options.adminEntraIds.split(',');
+          for (let i: number = 0; i < adminEntraIds.length; i++) {
```

Comment:
If you rebase with the latest version of `main`, we have a new utility function for this called `isValidUserPrincipalArray` or something like that.